src/Compilers/CSharp/Portable/Symbols/NamedTypeSymbol.cs

```diff
@@ -1510,9 +1510,18 @@ internal bool IsTupleTypeOfCardinality(out int tupleCardinality)
         }
 
         // PROTOTYPE: Should be abstract.
-        internal virtual NamedTypeSymbol AsNativeInt(bool asNativeInt) => this;
+        /// <summary>
+        /// Returns an instance of a symbol that has <see cref="IsNativeInt"/> set to <paramref name="asNativeInt"/>
+        /// if the underlying symbol represents System.IntPtr or System.UIntPtr.
+        /// For other symbols, throws <see cref="System.InvalidOperationException"/>.
+        /// </summary>
+        internal virtual NamedTypeSymbol AsNativeInt(bool asNativeInt) => throw ExceptionUtilities.Unreachable;
```

Comment:
Is it really useful to allow passing `asNativeInt: false`? The only place I could find could as easily just skip the call.

Then we wouldn't have the behavior of `NativeIntTypeSymbol.AsNativeInt` that can return the underlying type. That behavior strikes me as odd.

Reply:
I'll consider adding a `NativeIntUnderlyingType` property and removing the `bool asNativeInt` parameter in the follow up PR.

---
In reply to: [375000086](https://github.com/dotnet/roslyn/pull/40994#discussion_r375000086) [](ancestors = 375000086)